spec/rspec/expectations/block_snippet_extractor_spec.rb

```diff
@@ -0,0 +1,267 @@
+
+        it 'returns empty lines' do
+          expect(body_content_lines).to eq([])
+        end
+      end
+
+      context 'with `target_method { body }`' do
+        let(:expression) do
+          target_method { 1.positive? }
+        end
+
+        it 'returns the body content lines' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context 'with `target_method do body end`' do
+        let(:expression) do
+          target_method do 1.positive? end
+        end
+
+        it 'returns the body content lines' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context 'with `target_method { |arg1, arg2| body }`' do
+        let(:expression) do
+          target_method { |arg1, arg2| 1.positive? }
+        end
+
+        it 'returns the body content lines' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context 'with `target_method(:arg1, :arg2) { body }`' do
+        let(:expression) do
+          target_method(:arg1, :arg2) { 1.positive? }
+        end
+
+        it 'returns the body content lines' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context 'with `target_method(:arg1,:arg2){|arg1,arg2|body}`' do
+        let(:expression) do
+          target_method(:arg1,:arg2){|arg1,arg2|1.positive?}
+        end
+
+        it 'returns the body content lines' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context 'with a multiline block containing a single line body' do
+        let(:expression) do
+          target_method do
+            1.positive?
+          end
+        end
+
+        it 'returns the body content lines' do
+          expect(body_content_lines).to eq(['1.positive?'])
+        end
+      end
+
+      context 'with a multiline body block' do
+        let(:expression) do
+          target_method do
+            1.positive?
+            2.negative?
+          end
+        end
+
+        it 'returns the body content lines' do
+          expect(body_content_lines).to eq([
+            '1.positive?',
+            '2.negative?'
+          ])
+        end
+      end
+
+      context 'with `target_method { { :key => "value" } }`' do
+        let(:expression) do
+          target_method { { :key => "value" } }
+        end
+
+        it 'does not confuse the hash curly with the block closer' do
+          expect(body_content_lines).to eq(['{ :key => "value" }'])
+        end
+      end
+
+      context 'with a do-end block containing another do-end block' do
+        let(:expression) do
+          target_method do
+            2.times do |index|
+              puts index
+            end
+          end
+        end
+
+        it 'does not confuse the inner `end` with the outer `end`' do
+          expect(body_content_lines).to eq([
+            '2.times do |index|',
+            'puts index',
```

Comment:
Ideally it should properly handle indentations of multiline snippet, but for now we just use the result of BlockSnippetExtractor only when it's a single line. So I didn't implement the indentation handling since YAGNI.